sdks/java/io/elasticsearch/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTest.java

```diff
@@ -0,0 +1,302 @@
+    serverSocket.close();
+
+    FileUtils.deleteDirectory(new File(DATA_DIRECTORY));
+    LOGGER.info("Starting embedded Elasticsearch instance ({})", esHttpPort);
+    Settings.Builder settingsBuilder =
+        Settings.settingsBuilder()
+            .put("cluster.name", "beam")
+            .put("http.enabled", "true")
+            .put("node.data", "true")
+            .put("path.data", DATA_DIRECTORY)
+            .put("path.home", DATA_DIRECTORY)
+            .put("node.name", "beam")
+            .put("network.host", ES_IP)
+            .put("http.port", esHttpPort)
+            .put("index.store.stats_refresh_interval", 0);
+    node = NodeBuilder.nodeBuilder().settings(settingsBuilder).build();
+    LOGGER.info("Elasticsearch node created");
+    if (node != null) {
+      node.start();
+    }
+  }
+
+  @Before
+  public void before() throws Exception {
+    IndicesAdminClient indices = node.client().admin().indices();
+    IndicesExistsResponse indicesExistsResponse =
+        indices.exists(new IndicesExistsRequest(ES_INDEX)).get();
+    if (indicesExistsResponse.isExists()) {
+      indices.prepareDelete(ES_INDEX).get();
+    }
+  }
+
+  private void sampleIndex(long nbDocs) throws Exception {
+    Client client = node.client();
+    final BulkRequestBuilder bulkRequestBuilder = client.prepareBulk().setRefresh(true);
+
+    String[] scientists =
+        { "Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday", "Newton", "Bohr",
+            "Galilei", "Maxwell" };
+    for (int i = 0; i < nbDocs; i++) {
+      int index = i % scientists.length;
+      String source = String.format("{\"scientist\":\"%s\", \"id\":%d}", scientists[index], i);
+      bulkRequestBuilder.add(client.prepareIndex(ES_INDEX, ES_TYPE, null).setSource(source));
+    }
+    final BulkResponse bulkResponse = bulkRequestBuilder.execute().actionGet();
+    if (bulkResponse.hasFailures()) {
+      throw new IOException("Cannot insert samples in index " + ES_INDEX);
+    }
+  }
+
+  @Test
+  public void testSizes() throws Exception {
+    sampleIndex(NB_DOCS);
+    PipelineOptions options = PipelineOptionsFactory.create();
+    String[] addresses = {"http://" + ES_IP + ":" + esHttpPort};
+    ElasticsearchIO.Read read =
+        ElasticsearchIO.read().withConnectionConfiguration(
+            ElasticsearchIO.ConnectionConfiguration
+                .create(addresses, ES_INDEX, ES_TYPE));
+    BoundedElasticsearchSource initialSource =
+        new BoundedElasticsearchSource(read, null);
+    // can't use equal assert as Elasticsearch indexes never have same size
+    // (due to internal Elasticsearch implementation)
+    long estimatedSize = initialSource.getEstimatedSizeBytes(options);
+    LOGGER.info("Estimated size: {}", estimatedSize);
+    assertTrue("Wrong estimated size", estimatedSize > 40000);
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testRead() throws Exception {
+    sampleIndex(NB_DOCS);
```

Comment:
No because testWrite* tests do not need to insert a dataset because their point is to insert data.